### DISCUSSION TITLE
Upload image(s) to instruction steps

### DISCUSSION THREAD

**Discussion by cashpw** [mealie-recipes/mealie#885] [Feature Request]
**Before submitting this feature request I have**

- [x] Reviewed the documentation and verified that my feature request isn't already a feature.
- [x] Reviewed the existing feature requests and my feature has not been requested.
- [X] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**

Allow for users to upload photos into ingredient steps in the same way they can upload a photo for the overall recipe. This would include both uploading from their computer and uploading/downloading from a provided URL.

Note that users _can_ embed an image in an instruction step via markdown (eg: ` ![Image description](http://foo.com/url-of-photo)`). However, this (1) doesn't provide a way for users to add images they have locally and (2) leaves linked images from another website at risk of going 404 if the host site ever moves them.

**(Optional): Suggest A Solution**

Use the same UI as in the overall recipe uploader; allow for multiple images.

**Additional Information**

- [ ] If the feature is accepted I'm willing to submit a PR to provide this feature

**Comment by Tukks** [comment]
It would be a great addition,
Actually when you add photo in the ressources part you need to download file to see the photo.
And like this site web for example : https://ricette.giallozafferano.it/Pasta-e-patate-alla-napoletana.html
You have photos on each step

**Comment by Plevuus** [comment]
I noticed that this feature does exist, but it is not usable on a mobile phone. The phone, however is the device I use to make most images. Having a way to add extra photos (per step) from the phone would be very valuable.

**Reply by xarantolus** [reply]
I'm currently looking into this, it should be possible to use the already existing upload logic and add a new button in the three dots menu of a recipe step

**Reply by tekstacey** [reply]
I am not seeing where this feature does exist.
I only see an option to download a single image for the whole recipe, not for parts.
Where do I look for information on this feature?

**Reply by mdeeter** [reply]
Using the desktop browser to edit:
![image](https://github.com/mealie-recipes/mealie/assets/136902/62ac2651-6924-4cd4-bbe7-cd4d3e4d978a)

it adds the image tag like so:
![image](https://github.com/mealie-recipes/mealie/assets/136902/cb71ffcf-6a69-49bb-ad74-0c8acbb8cad6)

**Comment by mdeeter** [comment]
̶A̶d̶d̶i̶n̶g̶ ̶a̶ ̶n̶o̶t̶e̶ ̶h̶e̶r̶e̶ ̶a̶b̶o̶u̶t̶ ̶t̶h̶i̶s̶ ̶f̶e̶a̶t̶u̶r̶e̶.̶ ̶ ̶I̶t̶ ̶w̶o̶r̶k̶s̶ ̶f̶i̶n̶e̶ ̶f̶o̶r̶ ̶m̶e̶.̶.̶.̶ ̶b̶u̶t̶ ̶I̶'̶d̶ ̶r̶e̶a̶l̶l̶y̶ ̶l̶o̶v̶e̶ ̶t̶o̶ ̶u̶p̶l̶o̶a̶d̶ ̶v̶i̶a̶ ̶a̶ ̶U̶R̶L̶ ̶i̶n̶s̶t̶e̶a̶d̶ ̶o̶f̶ ̶h̶a̶v̶i̶n̶g̶ ̶t̶o̶ ̶p̶u̶s̶h̶ ̶u̶p̶ ̶a̶ ̶f̶i̶l̶e̶.̶

̶S̶u̶g̶g̶e̶s̶t̶e̶d̶ ̶c̶h̶a̶n̶g̶e̶:̶ ̶
̶M̶a̶k̶e̶ ̶c̶u̶r̶r̶e̶n̶t̶ ̶u̶p̶l̶o̶a̶d̶ ̶o̶p̶t̶i̶o̶n̶ ̶c̶a̶l̶l̶e̶d̶ ̶"̶U̶p̶l̶o̶a̶d̶ ̶I̶m̶a̶g̶e̶ ̶F̶i̶l̶e̶"̶ ̶a̶n̶d̶ ̶a̶n̶ ̶a̶n̶o̶t̶h̶e̶r̶ ̶o̶p̶t̶i̶o̶n̶ ̶f̶o̶r̶ ̶"̶U̶p̶l̶o̶a̶d̶ ̶I̶m̶a̶g̶e̶ ̶b̶y̶ ̶U̶R̶L̶"̶ ̶w̶h̶i̶c̶h̶ ̶s̶h̶o̶w̶ ̶a̶n̶ ̶i̶n̶p̶u̶t̶ ̶f̶o̶r̶ ̶a̶ ̶U̶R̶L̶.̶ ̶ ̶I̶t̶ ̶w̶o̶u̶l̶d̶ ̶d̶o̶w̶n̶l̶o̶a̶d̶ ̶t̶h̶e̶ ̶f̶i̶l̶e̶ ̶a̶n̶d̶ ̶s̶a̶v̶e̶ ̶i̶t̶ ̶l̶o̶c̶a̶l̶l̶y̶ ̶(̶j̶u̶s̶t̶ ̶l̶i̶k̶e̶ ̶t̶h̶e̶ ̶c̶u̶r̶r̶e̶n̶t̶ ̶i̶m̶a̶g̶e̶ ̶u̶p̶l̶o̶a̶d̶)̶.̶

Snap... I just noticed and tried the Drag-n-Drop.  That's exactly what I needed!
![image](https://github.com/mealie-recipes/mealie/assets/136902/2945e0f5-07bf-4987-b82e-ae9d1b97cd87)

Thank you!
